### DISCUSSION TITLE
DBeaver does not display sub-seconds from SELECT on PostgreSQL timestamp column

### DISCUSSION THREAD

**Discussion by troogyt** [dbeaver/dbeaver#11469] [Q&A] [answered]
#### System information: 
- Win7 x64 6.1
- DBeaver Version: 7.3.1.202012201616 org.jkiss.dbeaver.core.product (Community Edition)
- Java Version: Java(TM) SE Runtime Environment (build 1.8.0_261-b12)

#### Connection specification:
- PostgreSQL 12.2, compiled by Visual C++ build 1914, 64-bit
- JSQLParser library 3.2.0 com.github.jsqlparser
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
When running a SELECT query on a timestamp column containing date time with a resolution of milliseconds stored as TEXT, DBeaver does not show the milliseconds (to any decimal place) but only displays the timestamp down to a resolution of seconds.

Using PgAdmin4 does not produce this error, instead it is displayed with millisecond resolution as expected.

### DBeaver - no milliseconds shown:
![image](https://user-images.githubusercontent.com/41710820/103271443-62a65e80-4a0e-11eb-91df-76bd9a97647a.png)

### PgAdmin4 - milliseconds shown:
![image](https://user-images.githubusercontent.com/41710820/103271561-adc07180-4a0e-11eb-94e6-20d9672b3c90.png)

#### Steps to reproduce, if exist:
INSERT INTO mytable
(date_time)
VALUES('05.08.2011 19:00:43.712');

SELECT
  date_time,
  (date_time || 'GMT')::TIMESTAMPTZ AS ts_tz,
  (date_time || 'GMT')::TIMESTAMPTZ AT TIME ZONE 'GMT' AS ts_gmt
FROM mytable;

#### Include any warning/errors/backtraces from the logs
None generated.

**Comment by troogyt** [comment]
Following up, using EXTRACT seems to return milliseconds:

`SELECT date_time, EXTRACT(SECOND FROM date_time::TIMESTAMPTZ) millisecs FROM mytable;`

![image](https://user-images.githubusercontent.com/41710820/103353864-0e7aa780-4afe-11eb-9a3b-ab4117fce28c.png)

**Comment by kseniiaguzeeva** [comment]
Please send also the table DDL

**Comment by uslss** [comment]
there is no update on the issue for a long time.
if it is still actual for you - feel free to reopen it with an additional info

**Comment by troogyt** [comment]
Hi,

Sorry for the delay. I got busy.
The DDL for the table above is:

`CREATE TABLE public.mytable
(
    date_time text COLLATE pg_catalog."default" NOT NULL
)`

though what I really entered was:
`CREATE TABLE mytable
(
    date_time TEXT NOT NULL
)`

**Comment by troogyt** [comment]
@uslss

Sorry to ask, but how do I re-open this ticket?
Can't find obvious instructions on how to do so.

Thanks.

**Comment by p91paul** [comment]
Either some staff member reopens it, or you'll have to open a new one.

**Comment by troogyt** [comment]
Is there a way to 'step' debug the query as it performs the SELECT? Any debug logs I can look at?

**Comment by troogyt** [comment]
Thanks for re-opening, @kseniiaguzeeva.

**Comment by kseniiaguzeeva** [comment, marked as answer]
It is not a bug, you can set a date format in preferences>Editor>Data Editor>Data formats https://dbeaver.com/docs/wiki/Managing-Data-Formats/

**Comment by troogyt** [comment]
Thank you. It worked.
